### DISCUSSION TITLE
Limitations

### DISCUSSION THREAD

**Discussion by artjoma** [apache/kvrocks#1898] [Q&A]
Could you please advice. Where I can read about limitations ? For example, what is max count of keys in Map or max length(bytes) of value in Map ?

**Comment by git-hulk** [comment]
Hi @artjoma 

The max count of keys in Hash/List/Set/ZSet is 2^32 for now.

**Reply by artjoma** [reply]
Thank you! What about value? I know the max bulk string is 512Mb. Is it possible increase it for example till 4Gb ?

**Reply by wessorh** [reply]
I'm interested in storing 1Gb values and I'd like to better understand where in the code the limitation are. 
thanks!

**Reply by git-hulk** [reply]
Hi @wessorh 

You can find the related codes here: https://github.com/apache/kvrocks/blob/unstable/src/server/redis_request.cc#L111, and you can change the value and recompile if would like to support the larger values.
